nncf/openvino/engine.py

```diff
@@ -74,3 +70,43 @@ def infer(
             for tensor_name in tensor.get_names():
                 output_data[tensor_name] = value
         return output_data
+
+
+class OVNativeEngine(Engine):
+    """
+    Implementation of the engine for OpenVINO backend.
+
+    OVNativeEngine uses
+    [OpenVINO Runtime](https://docs.openvino.ai/latest/openvino_docs_OV_UG_OV_Runtime_User_Guide.html)
+    to infer the model.
+    """
+
+    def __init__(self, model: ov.Model, target_device: TargetDevice = TargetDevice.CPU):
+        self.engine = None
+        self.model = model
+        self.target_device = target_device
+        if target_device == TargetDevice.ANY:
+            self.target_device = TargetDevice.CPU
+
+    def _get_compiled_model(self) -> ov.CompiledModel:
+        """
+        Returns OpenVINO compiled model
+
+        :return: A compiled model
+        """
+        return ov.Core().compile_model(self.model, self.target_device.value)
+
+    def infer(
+        self, input_data: Union[np.ndarray, List[np.ndarray], Tuple[np.ndarray], Dict[str, np.ndarray]]
+    ) -> Dict[str, np.ndarray]:
+        """
+        Runs model on the provided input via OpenVINO Runtime.
+        Returns the dictionary of model outputs by node names.
+
+        :param input_data: Inputs for the model.
+        :return output_data: Model's output.
+        """
+        if self.engine is None:
+            compiled_model = self._get_compiled_model()
```

Comment:
Do we really need `_get_compiled_model()` method? Why it is better to use method instead of
```python
compiled_model = ov.Core().compile_model(self.model, self.target_device.value)
```

Reply:
Can I assign the same question to you?

Reply:
Sure! I don't see any reason why this method is required here. So I would like to understand your point of view. This is the reason why I asked the question.